script.js

```diff
@@ -0,0 +1,157 @@
+
+const userEntry = document.querySelector('#display');
+const btnContainer = document.querySelector('#button-container');
+const operBtns = document.querySelectorAll('.oper');
+const backspace = document.querySelector('#backspace');
+
+//handles keyboard entries
+window.addEventListener('keydown', event => {
+    if (!isNaN(event.key)) userEntry.textContent += event.key;
+
+    if (oper.hasOwnProperty(event.key)) {
+        userEntry.textContent += event.key;
+    };
+
+    if (event.key === '/') userEntry.textContent += '÷';
+
+    if (event.key === 'Enter') {
+        const result = shuntingYard(userEntry.textContent);
+        userEntry.textContent = result;
+    };
+});
```

Comment:
if
else if
else if

the event key wont be equally divide and enter during the same run through of the listener

Reply:
veeery minor detail but line 30- not entirely clear what youre attempting to do with the slice operation, store it in a variable which indicates its purpose and then assign it to text content

otherwise, this is resolved and generally looks a lot cleaner- if you wanted to, you could implement a switch case and have, for example: 

case 'enter'
case 'Escape'
case 'backspace
default
//iterate through buttons 

with its current structure i would reshuffle the checks: 

if 'enter'
else if 'Escape'
else if 'backspace
else
//iterate through buttons 

instead of iterating and checking all the buttons before the if. When the event corresponds to a button, you will need to iterate through them, however if they click escape you are currently checking lengthOfButtonArray+1 times

while in this case its a micro optimisation and not necessary at all, its better to have your heaviest operations run only when necessary as opposed to every single time 

also, quick tip: 

 buttonList.some(button => {
        if (key !== button.value) return false
            userEntry.push(key);
            (key === '/')
            ? display.textContent += '÷'
             : display.textContent += key;
            return true
        };
    });
    
    the some operation bails early when an element is found matching true on whatever condition you pass it, suppose we have an array of 100 elements, and the very first element is the one that meets the condition- 
    
    //this will iterate 100 times despite the first element being the desired one
    array.forEach((element)=>something)
    
    //this will iterate once 
    array.some((element)=>something)

   theres also array.every which bails out when an element returns false, good for checking if everything is valid in an array and throwing errors if not, or permission checking and restricting access